crates/uv-git/src/source.rs

```diff
@@ -71,74 +72,100 @@ impl GitSource {
 
     /// Fetch the underlying Git repository at the given revision.
     #[instrument(skip(self), fields(repository = %self.git.repository(), rev = ?self.git.precise()))]
-    pub fn fetch(self) -> Result<Fetch> {
+    pub async fn fetch(self, keyring_provider: Option<&KeyringProvider>) -> Result<Fetch> {
```

Comment:
This method was switched to `async` in order to call `KeyringProvider::fetch_if_native`. As a result, a immediately invoked closure below was converted to a `spawn_blocking` call. It's worth noting that this change holds whether or not the keyring provider is configured to be native.